client/pages/Dashboard/index.jsx

```diff
@@ -116,7 +131,14 @@ class Dashboard extends Component {
    */
   render() {
     const { location: { search } } = this.props;
-    const { activeUser } = this.state;
+    const { activeUser, categoryList } = this.state;
+
+    const categories = Array.isArray(categoryList) ? categoryList.map(cat => ({
```

Comment:
Nice work. Now that the category list is already initialized as an array. Do we need to check if its an array ?

Reply:
@kenneth-kip yes. The implementation from backend sometimes returns empty object {}.

Reply:
Aha. Make sense

Reply:
One last thing. Instead of `cat` use `category`